Source/debug.cpp

```diff
@@ -58,32 +55,38 @@ void GiveGoldCheat()
 {
 	auto &myPlayer = plr[myplr];
 
-	for (int i = 0; i < NUM_INV_GRID_ELEM; i++) {
-		if (myPlayer.InvGrid[i] == 0) {
-			int ni = myPlayer._pNumInv++;
-			SetPlrHandItem(&myPlayer.InvList[ni], IDI_GOLD);
-			GetPlrHandSeed(&myPlayer.InvList[ni]);
-			myPlayer.InvList[ni]._ivalue = GOLD_MAX_LIMIT;
-			myPlayer.InvList[ni]._iCurs = ICURS_GOLD_LARGE;
-			myPlayer._pGold += GOLD_MAX_LIMIT;
-			myPlayer.InvGrid[i] = myPlayer._pNumInv;
-		}
+	for (int8_t &itemId : myPlayer.InvGrid) {
+		if (itemId != 0)
+			continue;
+
+		int ni = myPlayer._pNumInv++;
+		SetPlrHandItem(&myPlayer.InvList[ni], IDI_GOLD);
+		GetPlrHandSeed(&myPlayer.InvList[ni]);
+		myPlayer.InvList[ni]._ivalue = GOLD_MAX_LIMIT;
+		myPlayer.InvList[ni]._iCurs = ICURS_GOLD_LARGE;
+		myPlayer._pGold += GOLD_MAX_LIMIT;
+		itemId = myPlayer._pNumInv;
 	}
 }
 
 void TakeGoldCheat()
 {
 	auto &myPlayer = plr[myplr];
 
-	for (int i = 0; i < NUM_INV_GRID_ELEM; i++) {
-		int8_t ig = myPlayer.InvGrid[i];
-		if (ig > 0 && myPlayer.InvList[ig - 1]._itype == ITYPE_GOLD)
-			myPlayer.RemoveInvItem(ig - 1);
+	for (int8_t itemId : myPlayer.InvGrid) {
+		itemId -= 1;
+
+		if (itemId < 0)
+			continue;
+		if (myPlayer.InvList[itemId]._itype != ITYPE_GOLD)
+			continue;
+
+		myPlayer.RemoveInvItem(itemId);
 	}
 
-	for (int i = 0; i < MAXBELTITEMS; i++) {
-		if (myPlayer.SpdList[i]._itype == ITYPE_GOLD)
-			myPlayer.SpdList[i]._itype = ITYPE_NONE;
+	for (auto &item : myPlayer.SpdList) {
+		if (item._itype == ITYPE_GOLD)
+			item._itype = ITYPE_NONE;

```

Comment:
Why is there all this code that checks for gold in the player's belt? Can we remove this?

Reply:
Gold was at some point allowed to go in the belt. There are a few places in the codebase that check this. Would suggest leaving to a separate change that focuses on that aspect alone.

Reply:
I think most of it has already been deleted else where.

Reply:
@julealgon Did they patch it out, or was it changed before the official release?